src/main/java/com/minecolonies/coremod/client/gui/WindowHomeBuilding.java

```diff
@@ -1,24 +1,151 @@
+
+    /**
+     * The list of citizen assigned to this hut.
+     */
+    private ScrollingList citizen;
+
+    /**
+     * The building the view is relates to.
+     */
+    private final BuildingHome.View home;
+
+    /**
      * Creates the Window object.
      *
      * @param building View of the home building.
      */
     public WindowHomeBuilding(final BuildingHome.View building)
     {
         super(building, Constants.MOD_ID + HOME_BUILDING_RESOURCE_SUFFIX);
+
+        super.registerButton(BUTTON_ASSIGN, this::assignClicked);
+        super.registerButton(BUTTON_REMOVE, this::removeClicked);
+        this.home = building;
+    }
+
+    /**
+     * Refresh the view.
+     */
+    private void refreshView()
+    {
+        final Button button = findPaneOfTypeByID(BUTTON_ASSIGN, Button.class);
+        final int sparePlaces = building.getBuildingLevel() - building.getResidents().size();
+        button.setLabel(LanguageHandler.format(COM_MINECOLONIES_COREMOD_GUI_HOME_ASSIGN, sparePlaces));
+        if (sparePlaces <= 0)
+        {
+            button.disable();
+        }
+        else
+        {
+            button.enable();
+        }
+        citizen.refreshElementPanes();
+    }
+
+    @Override
+    public void onOpened()
+    {
+        super.onOpened();
+
+        citizen = findPaneOfTypeByID(LIST_CITIZEN, ScrollingList.class);
+        citizen.setDataProvider(new ScrollingList.DataProvider()
+        {
+            @Override
+            public int getElementCount()
+            {
+                return home.getResidents().size();
+            }
+
+            @Override
+            public void updateElement(final int index, @NotNull final Pane rowPane)
+            {
+                rowPane.findPaneOfTypeByID("name", Label.class).setLabelText(home.getColony().getCitizen(home.getResidents().get(index)).getName());
+            }
+        });
+
+        refreshView();
+    }
+
+    /**
+     * Action when an assign button is clicked.
+     *
+     * @param button the clicked button.
+     */
+    private void assignClicked(@NotNull final Button button)
+    {
+        if (building.getColony().isManualHousing())
+        {
+            if (building.getBuildingLevel() == 0)
+            {
+                LanguageHandler.sendPlayerMessage(Minecraft.getMinecraft().thePlayer, "com.minecolonies.coremod.gui.workerHuts.level0");
+                return;
+            }
+
+            if (building.getResidents().size() < building.getBuildingLevel())
+            {
+                @NotNull final WindowAssignCitizen window = new WindowAssignCitizen(building.getColony(), building.getLocation());
+                window.open();
+            }
+        }
+    }
+
+    /**
+     * Action when the remove button is clicked.
+     *
+     * @param button the clicked button.
+     */
+    private void removeClicked(@NotNull final Button button)
+    {
+        if (building.getColony().isManualHousing())
```

Comment:
We need to let the player know that the remove button does nothing or/ disable it when isManualHousing is false